pkg/backend/httpstate/client/client.go

```diff
@@ -546,22 +553,36 @@ func (pc *Client) PublishPolicyPack(ctx context.Context, orgName string,
 	// publishing the PolicyPack.
 	//
 
+	// If a version tag was provided, validate version tag before sending the request. If no tag was provided,
+	// likely its an older version of pulumi/policy that does not implement version tags.
+	if analyzerInfo.VersionTag != "" && !policyPackVersionTagRE.MatchString(analyzerInfo.VersionTag) {
+		msg := fmt.Sprintf("invalid version %q - version may only contain alphanumeric, hyphens, underscores, "+
+			"periods, or colons. It must also be between 1 and 150 characters long.", analyzerInfo.VersionTag)
+		return 0, errors.New(msg)
+	}
+
 	req := apitype.CreatePolicyPackRequest{
 		Name:        analyzerInfo.Name,
 		DisplayName: analyzerInfo.DisplayName,
+		VersionTag:  analyzerInfo.VersionTag,
 		Policies:    analyzerInfo.Policies,
 	}
 
-	fmt.Printf("Publishing %s to %s\n", analyzerInfo.Name, orgName)
+	// Print a publishing message. We have to handle the case where an older version of pulumi/policy
+	// is in use, which does not provide  a version tag.
+	publishingMsg := fmt.Sprintf("Publishing %q - version %s to %q organization\n",
+		analyzerInfo.Name, analyzerInfo.VersionTag, orgName)
+	if analyzerInfo.VersionTag == "" {
+		publishingMsg = fmt.Sprintf("Publishing %q to %q organization\n", analyzerInfo.Name, orgName)
+	}
+	fmt.Print(publishingMsg)
```

Comment:
This seems problematic. We shouldn't have the Pulumi API client also deal with writing output messages directly to STDOUT. This makes it impossible to replicate the same output on the Pulumi Service, since the message isn't part of the engine event stream.

So at the very least, we should have the caller (or someone even lower in the callstack) call `fmt.Print`. And even then, we would want to avoid calling `fmt.Print` directly, but instead emit a diagnostic event.

Reply:
Agreed -- although this is how it was initially implemented. I thought we had an open issue to address this, but alas I dont see it. If you think it's critical, please open another issue.